sharktank/sharktank/layers/configs/llm_configs.py

```diff
@@ -173,32 +103,19 @@ def from_gguf_props(p: dict[str, Any]):
                 p, f"{name_prefix}.attention.head_count_kv", attention_head_count
             ),
             attn_head_dim=attn_head_dim,
-            q_lora_rank=q_lora_rank,
-            kv_lora_rank=kv_lora_rank,
-            qk_nope_head_dim=qk_nope_head_dim,
-            qk_rope_head_dim=qk_rope_head_dim,
-            v_head_dim=v_head_dim,
-            attention_softcap=attention_softcap,
             expert_count=expert_count,
             expert_used_count=_optional_int_prop(
                 p, f"{name_prefix}.expert_used_count", default_expert_used_count
             ),
             moe_intermediate_size=_optional_int_prop(
                 p, f"{name_prefix}.moe_intermediate_size", None
             ),
-            route_scale=route_scale,
             n_dense_layers=n_dense_layers,
-            expert_shared_count=expert_shared_count,
-            n_expert_groups=n_expert_groups,
-            n_limited_groups=n_limited_groups,
             rope_dimension_count=rope_dimension_count,
             rope_freq_base=_optional_float_prop(
                 p, f"{name_prefix}.rope.freq_base", default_rope_freq_base
             ),
-            rope_scaling_type=rope_scaling_type,
-            rope_scaling_factor=rope_scaling_factor,
-            rope_scaling_original_context_length=rope_scaling_original_context_length,
-            rope_scaling_yarn_log_multiplier=rope_scaling_yarn_log_multiplier,
+            custom=custom_configs,
```

Comment:
So here I would do `**custom_config` and still have the named values in your `LLaMaHParams` this way you can still have validationg that the names match to params. Its just a nice trick that allows you to optionally populate args.